invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -57,7 +57,7 @@ class CSLJSONSchema(Schema):
     language = fields.Method("get_language")
     version = SanitizedUnicode(attribute="metadata.version")
     note = fields.Method("get_note")
-    doi = fields.Method("get_doi", data_key="DOI")
+    doi = fields.Str(attribute="pids.doi.identifier", data_key="DOI")
```

Comment:
Small question: are we sure that the logic of the `missing` field or something similar stays if we use it like this?

Reply:
Yes, the need for `missing` is because if a `Method` field returns `None` it will be use as its value, so there is a need to specify "ignore this field". For example:

```python
>>> from marshmallow import Schema, fields, missing

>>> class A(Schema):
...     test = fields.Str()
...     another = fields.Str()

>>> data = {"test": 1}
>>> A().dump(data)
{'test': '1'}       < --- See here the key "another" is not present
>>> A().dump({})
{}                     < --- Both keys are ignored
```

Now if we replace test by a method:

```python
>>> class B(Schema):
...     test = fields.Method("get_test")
...     another = fields.Str()
...     def get_test(self, obj):
...             return obj.get("test")
...
>>> B().dump(data)
{'test': 1}
>>> B().dump({})
{'test': None}     <--- we got `None` as value
```

Now we if return missing we go back to the same behaviour than in the first case:

```python
>>> class C(Schema):
...     test = fields.Method("get_test")
...     another = fields.Str()

...     def get_test(self, obj):
...             return obj.get("test", missing)
...
>>> C().dump(data)
{'test': 1}
>>> C().dump({})
{}
```